tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/label/LabelChangeRepository.scala

```diff
@@ -8,132 +8,94 @@ import org.apache.james.jmap.api.model.AccountId
 import org.reactivestreams.Publisher
 import reactor.core.scala.publisher.{SFlux, SMono}
 
-import java.time.ZonedDateTime
+import java.time.{Clock, Instant}
 import javax.inject.Named
 
 trait LabelChangeRepository {
-
-
   def save(labelChange: LabelChange): Publisher[Void]
 
   def getSinceState(accountId: AccountId, state: State, maxIdsToReturn: Option[Limit] = None): Publisher[LabelChanges]
 
   def getLatestState(accountId: AccountId): Publisher[State]
 
-  def getLatestStateWithDelegation(accountId: AccountId): Publisher[State]
-
-  def getSinceStateWithDelegation(accountId: AccountId, state: State, maxIdsToReturn: Option[Limit] = None): Publisher[LabelChanges]
 }
 
 case class LabelChange(accountId: AccountId,
                        created: Set[LabelId],
                        updated: Set[LabelId],
                        destroyed: Set[LabelId],
-                       newState: State,
-                       hasMoreChanges: Boolean,
-                       isDelegated: Boolean,
-                       date: ZonedDateTime) extends JmapChange {
+                       state: State) extends JmapChange {
   override def getAccountId: AccountId = accountId
 }
 
-case class LabelChanges(newState: State,
-                        hasMoreChanges: Boolean,
-                        created: Set[LabelId],
-                        updated: Set[LabelId],
-                        destroyed: Set[LabelId]) {
-  def getAllChanges: Set[LabelId] = created ++ updated ++ destroyed
-
-  def append(labelChange: LabelChange): LabelChanges =
-    this.copy(created = created ++ labelChange.created,
-      updated = updated ++ labelChange.updated,
-      destroyed = destroyed ++ labelChange.destroyed)
-}
-
-
-case class LabelChangesBuilder(var created: Set[LabelId] = Set(),
-                               var updated: Set[LabelId] = Set(),
-                               var destroyed: Set[LabelId] = Set(),
-                               var hasMoreChanges: Boolean = false,
-                               var canAppendMoreItem: Boolean = true,
-                               var state: State,
-                               limit: Int) {
+case class LabelChanges(limit: Int,
+                        private var _created: Set[LabelId] = Set(),
+                        private var _updated: Set[LabelId] = Set(),
+                        private var _destroyed: Set[LabelId] = Set(),
+                        private var _hasMoreChanges: Boolean = false,
+                        private var _canAppendMoreItem: Boolean = true,
+                        private var _newState: State) {
```

Comment:
Don't use var

Reply:
even if it has been declared private?
I use it for collectReduce function

Reply:
yes